resources/qml/PrinterSelector/MachineListButton.qml

```diff
@@ -7,81 +7,134 @@ import QtQuick.Controls 2.3
 import UM 1.5 as UM
 import Cura 1.0 as Cura
 
-
-Button
-{
-    id: machineListButton
-
+Loader {
+    id: loader
     width: parent.width
-    height: UM.Theme.getSize("large_button").height
-    leftPadding: UM.Theme.getSize("default_margin").width
-    rightPadding: UM.Theme.getSize("default_margin").width
-    checkable: true
-    hoverEnabled: true
+    height: childrenRect.height
+    sourceComponent: {
+        switch (model.listType) {
+            case "HIDE_BUTTON":
+                hideButtonComponent
+                break;
+            case "SHOW_BUTTON":
+                showButtonComponent
+                break;
+            case "MACHINE":
+                machineListButtonComponent
+                break;
+            default:
+        }
+    }
+    property var onClicked
 
-    contentItem: Item
+    Component
     {
-        width: machineListButton.width - machineListButton.leftPadding - machineListButton.rightPadding
-        height: UM.Theme.getSize("action_button").height
-
-        UM.ColorImage
+        id: hideButtonComponent
```

Comment:
I wonder if it would be easier to just have a single show/hide (toggle) button component.

Reply:
I see what you mean. I think having two components with different text is easier to understand than having one component with swapping text.